fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3328,24 +3328,41 @@ void DDTeamCollection::updateCpuPivots() {
 }
 
 void DDTeamCollection::updateTeamEligibility() {
+	int healthyCount = 0, lowDiskUtilOnly = 0, lowCpuOnly = 0, lowDiskUtilLowCpu = 0;
 	for (auto& team : teams) {
 		if (team->isHealthy()) {
+			healthyCount++;
 			if (team->hasHealthyAvailableSpace(teamPivots.pivotAvailableSpaceRatio)) {
+				lowDiskUtilOnly++;
 				team->increaseEligibilityCount(data_distribution::EligibilityCounter::LOW_DISK_UTIL);
 			} else {
 				team->resetEligibilityCount(data_distribution::EligibilityCounter::LOW_DISK_UTIL);
 			}
 
 			if (team->hasLowerCpu(teamPivots.pivotCPU)) {
+				lowCpuOnly++;
 				team->increaseEligibilityCount(data_distribution::EligibilityCounter::LOW_CPU);
 			} else {
 				team->resetEligibilityCount(data_distribution::EligibilityCounter::LOW_CPU);
 			}
+
+			if (team->hasHealthyAvailableSpace(teamPivots.pivotAvailableSpaceRatio) &&
+			    (team->hasLowerCpu(teamPivots.pivotCPU))) {
+				lowDiskUtilLowCpu++;
+			}
```

Comment:
It's unnecessary to calculate the boolean again, you can reuse the bool result above by assigning to result to boolean variables.
And I don't think we should also count `lowDiskUtilLowCpu`. This kind of implies every time we add a new eligibility type, we need to add a count for the new combined type. I think we just need the worst case, saying no overlap of low disk util and low CPU, and make sure `lowCpuOnly + lowDiskUtilOnly < healthyCount`